pkg/kubelet/status/status_manager.go

```diff
@@ -352,12 +352,22 @@ func (m *manager) SetPodStatus(pod *v1.Pod, status v1.PodStatus) {
 	// Make sure we're caching a deep copy.
 	status = *status.DeepCopy()
 
+	// Set the observedGeneration for this pod status. Static pods are excluded.
+	if podObservedGenerationTrackingEnabled(pod) && !metav1.HasAnnotation(pod.ObjectMeta, v1.MirrorPodAnnotationKey) {
```

Comment:
I don't see a good reason to do special things for static pods... the less exceptional we can make them, the better, I think

https://github.com/kubernetes/enhancements/pull/5068#discussion_r1945434600 was the KEP thread on this, and https://github.com/kubernetes/kubernetes/pull/130181#discussion_r1956715409 was the metadata.generation comment

Reply:
Oh sorry, I misinterpreted your comment as suggesting that only in the api server we will not special-case mirror pods. I wrote in the KEP that the kubelet will explicitly exclude them, but TBH I think you are right that it doesn't really matter that much if we special-case them... just means that their generation/observedGeneration will always be on mirror pods 1 because of the way that they are implemented. I'm not sure how strict we are about following exactly what was written in the KEP, but I took out the mirror pod special-casing here

Reply:
> I'm not sure how strict we are about following exactly what was written in the KEP, but I took out the mirror pod special-casing here

It's normal to collect a few tweaks during implementation, just do a follow-up to update the KEP with any differences so it reflects what we actually implemented